### PR TITLE
OpenAPI: check super class for inherited auto-security resource methods

### LINKED ISSUE #40839: Extending a resource class loses OpenAPI security scheme when generated
### Describe the bug

When a resource extends an abstract class it will generate the OpenAPI methods but loses the security scheme.

```
components:
  securitySchemes:
    SecurityScheme:
      type: http
      description: Authentication
      scheme: bearer
      bearerFormat: Opaque
```

### Expected behavior

An extended resource extends the security scheme

```yaml
  /goodbye:
    get:
      tags:
      - Regular Resource
      responses:
        "200":
          description: OK
          content:
            text/plain:
              schema:
                type: string
        "403":
          description: Not Allowed
        "401":
          description: Not Authorized
      security:
      - SecurityScheme: []
```

### Actual behavior

The security scheme is missing from the generated OpenAPI spec.

```yaml
  /hello:
    get:
      tags:
      - Extended Resource
      responses:
        "200":
          description: OK
          content:
            text/plain:
              schema:
                type: string
```

### How to Reproduce?

1 . Run `mvn install`
2. See the `target/openapi.yml` 

[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/15443343/code-with-quarkus.zip)

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

21

### Quarkus version or git rev

3.10.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.9

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by MikeEdgar**
Fixes #40839

### CODE REVIEW

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java [hunk 1]
```diff
@@ -782,8 +796,17 @@ private static List<MethodInfo> getMethods(ClassInfo declaringClass, IndexView i
...
+        DotName superClassName = declaringClass.superName();
+        if (superClassName != null) {
+            ClassInfo superClass = index.getClassByName(superClassName);
+
+            if (superClass != null) {
+                methods.addAll(getMethods(superClass, index));
+            }
+        }
+
+        return methods;
```
Comment (gsmet) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:809]: Should we handle the case with a deeper hierarchy?

Comment (MikeEdgar) [reply extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:809]: Line 805 here recurses into the same method up until `java.lang.Object`. Granted, there is no test case for multi-level, but I can add that.

Comment (MikeEdgar) [reply extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:809]: I've added a multi-level resource class hierarchy to the existing test case.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**